File: BC-Free/app/src/main/java/ca/josephroque/bowlingcompanion/utilities/DateUtils.java
package ca.josephroque.bowlingcompanion.utilities;

import android.util.Log;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

/**
 * Created by Joseph Roque on 2015-07-28. Provides methods for managing dates and times as milliseconds since January 1,
 * 1970, at 00:00:00 GMT, as {@link java.util.Date} objects, or as {@link java.util.Calendar} objects.
 */
public final class DateUtils {

    /** Identifies output from this class in Logcat. */
    @SuppressWarnings("unused")
    private static final String TAG = "DateUtils";

    /** A value in milliseconds equivalent to exactly one week. */
    public static final long MILLIS_ONE_WEEK = 1000 * 60 * 60 * 24 * 7;

    /**
     * Formats a String of the format 'yyyy-MM-dd HH:mm:ss' to a {@link java.util.Date} object.
     */
    private static final DateFormat STRING_TO_DATE = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss",
            Locale.CANADA);

    /**
     * Parses a String to a {@link java.util.Date}.
     *
     * @param strDate date as a string in the format 'yyyy-MM-dd HH:mm:ss'
     * @return the date represented by the value in the column
     */
    public static Date parseEntryDate(String strDate) {
        Date date;
        try {
            date = STRING_TO_DATE.parse(strDate);
        } catch (ParseException ex) {
            Log.e(TAG, "Unable to parse entry date", ex);
            return null;
        }

        return date;
    }

    /**
     * Gets a new {@link java.util.Calendar} instance set to midnight on the date of {@code date}.
     *
     * @param date date for calendar
     * @return new instance of {@code Calendar}.
     */
    public static Calendar getCalendarAtMidnight(Date date) {
        Calendar calendar = Calendar.getInstance(Locale.CANADA);
        calendar.setTime(date);
        calendar.set(Calendar.HOUR, 0);
        calendar.set(Calendar.MINUTE, 0);
        calendar.set(Calendar.SECOND, 0);
        calendar.set(Calendar.MILLISECOND, 0);

        return calendar;
    }

    /**
     * Default private constructor.
     */
    private DateUtils() {
        // does nothing
    }
}

File: legacy/app/src/main/java/ca/josephroque/bowlingcompanion/utils/DateUtils.kt
package ca.josephroque.bowlingcompanion.utils

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

/**
 * Copyright (C) 2018 Joseph Roque
 *
 * Utility methods for date manipulation and display.
 */
object DateUtils {

    // MARK: DateUtils

    /**
     * Convert a series date [String] to a [Date].
     *
     * @param seriesDate date to format
     * @return a date object
     */
    fun seriesDateToDate(seriesDate: String): Date {
        val formatter = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.CANADA)
        return formatter.parse(seriesDate)
    }

    /**
     * Convert a [Date] to a series date [String].
     *
     * @param date date to format
     * @return a string suitable for a series
     */
    fun dateToSeriesDate(date: Date): String {
        val formatter = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.CANADA)
        return formatter.format(date)
    }

    /**
     * Converts [Date] to a cleaner format.
     *
     * @param date date to format
     * @return prettier format of string with full month name
     */
    fun dateToPretty(date: Date): String {
        val formatter = SimpleDateFormat("MMMM dd, yyyy", Locale.CANADA)
        return formatter.format(date)
    }

    /**
     * Converts [Date] to a shorter format.
     *
     * @param date date to format
     * @return shorter format of string with month and day
     */
    fun dateToShort(date: Date): String {
        val formatter = SimpleDateFormat("MM/dd", Locale.CANADA)
        return formatter.format(date)
    }
}
